### DISCUSSION TITLE
multiple ui.download in one page

### DISCUSSION THREAD

**Discussion by bobanovo** [zauberzeug/nicegui#2970] [Q&A]
### Question

Hi, all, 
pls, how do this use case - on one page offer more ui.download butotns for download different files or content?

I have this code:

```
from nicegui import app, ui, Tailwind, Client

import uvicorn
from fastapi import FastAPI
from fastapi.staticfiles import StaticFiles

app = FastAPI()
app.mount("/static", StaticFiles(directory="."), name="static")


arrayfile=['exposed-services.yaml','internal-services.yaml','int-cert-manager.yaml']

for f in arrayfile:
    print(f)
    ui.button(f,icon='download', on_click=lambda: ui.download('static/'+f,f)).props('fab color=orange').classes('self-center')

ui.run()
```

but, when I click to any button, still download only last file... What is wrong in my code/head? :-)

thanks a lot, Radek

**Comment by rodja** [comment]
This has been asked many many times. Have a look into our FAQs: https://github.com/zauberzeug/nicegui/wiki/FAQs#why-have-all-my-elements-the-same-value

**Reply by bobanovo** [reply]
ahhh, ok, a big sorry!

**Reply by bobanovo** [reply]
thanks!
